data_science/palette/notebooks/00 - RGB, LAB, and human colour perception.ipynb

```diff
@@ -0,0 +1,133 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# RGB and CIE\n",
+    "In this project we'll regularly need to calculate distances between pixels in colour space as a proxy for the visual difference between the colours. The simplest way of doing this is to calculate the cosine/euclidian/l2 distance, as written below in maths and python:\n",
+    "\n",
+    "${\\displaystyle \\Delta C ={\\sqrt {(R_{2}-R_{1})^{2}+(G_{2}-G_{1})^{2}+(B_{2}-B_{1})^{2}}}}$"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def colour_distance(colour_1, colour_2):\n",
+    "    return sum([(a - b) ** 2 for a, b in zip(colour_1, colour_2)]) ** 0.5"
```

Comment:
minor: Maybe name `a` and `b` in such a way that emphasises the connection to $C_1$ and $C_2$? How about:

```python
return sum([(comp_2 - comp_1) ** 2 for comp_1, comp_2 in zip(colour_1, colour_2)]) ** 0.5
```

(Note that I’ve also flipped the order to match your mathematical definition. I know it’s the same calculation, but it aids readability.)